### PR TITLE
Feature/graphql setup

### PR DESCRIPTION
**Pull request by lesser1**
Initial PR for the graphql setup.

### CODE REVIEW

File: src/views/ProductView.vue [hunk 1]
```diff
@@ -33,45 +33,53 @@
...
                         <v-card-subtitle>{{
-                            productVariantName
+                            productVariant?.currentVersion.name
                         }}</v-card-subtitle>
                     </v-card-item>
                     <v-card-text>
                         <div class="d-flex flex-wrap ga-2">
-                            <v-chip>Clothing</v-chip>
-                            <v-chip>T-Shirts</v-chip>
+                            <v-chip>Dummy Clothing</v-chip>
```
Comment (nk-coding) [inline src/views/ProductView.vue:45]: Are here still dummy values required?

File: src/views/ProductView.vue [hunk 2]
```diff
@@ -33,45 +33,53 @@
...
                         <v-select
-                            :items="['S', 'M', 'L']"
+                            :items="['Dummy S', 'Dummy M', 'Dummy L']"
                             density="compact"
                             label="Size"
                             variant="outlined"
                         ></v-select>
                     </div>
-                    <v-card-actions>
+                    <!-- <v-card-actions>
```
Comment (nk-coding) [inline src/views/ProductView.vue]: can commented out code be removed?

File: temp-schema.graphql
Comment (nk-coding) [inline temp-schema.graphql:1]: maybe removing this and then also pointing the codegen file to the api is a good idea

**Review by nk-coding — CHANGES_REQUESTED** [review]
Also one could think about adding some code doc

File: src/views/ProductView.vue (hunk 1 above)
Comment (lesser1) [reply src/views/ProductView.vue:45]: I think at that point there were still no categories available from the service. Adding the categories to the frontend has to happen next, of course.

File: src/views/ProductView.vue (hunk 2 above)
Comment (lesser1) [reply src/views/ProductView.vue]: Sure, will do that immediately.

File: src/views/ProductView.vue (hunk 1 above)
Comment (lesser1) [reply src/views/ProductView.vue:45]: Imo, everything related to the categories should be added via a dedicated PR in the near future.

**Review by nk-coding — CHANGES_REQUESTED** [review]
are you sure you ran the formatter?

**Comment by lesser1** [comment]
> are you sure you ran the formatter?

Forgot that... smh. Will do...

File: src/views/ProductView.vue [hunk 3]
```diff
@@ -104,19 +108,47 @@
...
+import { asyncComputed } from '@vueuse/core'
+import { useClient } from '@/graphql/client'
+import { useRoute } from 'vue-router'
+import { computed } from 'vue'
+
+const client = useClient()
+const route = useRoute()
+
+// get the product id from the route params
+const id = route.params.productid.toString()
```
Comment (nk-coding) [inline src/views/ProductView.vue]: I'm pretty sure one should use `computed` here

File: src/views/ProductView.vue [hunk 4]
```diff
@@ -104,19 +108,47 @@
...
+
+// see if the route also contains a product variant id
+const productVariantId = route.params.productvariantid?.toString()
```
Comment (nk-coding) [inline src/views/ProductView.vue]: here too

**Review by nk-coding — CHANGES_REQUESTED** [review]

**Review by nk-coding — APPROVED** [review]

**Pull request merged**